### PR TITLE
Fix asyncomplete-lsp#15, asyncomplete#133

### PR DESCRIPTION
**Pull request by yaegassy**
fix.

* https://github.com/prabirshrestha/asyncomplete-lsp.vim/issues/15
* https://github.com/prabirshrestha/asyncomplete.vim/issues/133

### CODE REVIEW

File: plugin/asyncomplete-lsp.vim [hunk 1]
```diff
@@ -11,7 +11,7 @@ au User lsp_server_exit call s:server_exited()
 function! s:server_initialized() abort
     let l:server_names = lsp#get_server_names()
     for l:server_name in l:server_names
-        if !has_key(s:servers, l:server_name)
+        if !has_key(s:servers, l:server_name) || s:servers[l:server_name] == 0
```
Comment (prabirshrestha) [inline plugin/asyncomplete-lsp.vim]: Can we use empty instead of 0 check?

**Comment by prabirshrestha** [comment]
@yaegassy Can you let me know why we would need that check and underwhat scenario it would hit that code path as we are setting whitelist for the filetype.

@maamo888 Can you try this branch and see if it works.

I tried with tsx files for react with buffer, syntax and typescript-language-server and it seems to work without this PR. Note: language server do takes time to start so could be that issue. Can you try if `:LspDefinition` works. A log would also be useful here.

**Comment by prabirshrestha** [comment]
I understand the issue now. Seems like it would work. another option would be to remove the else block to set it to 0 so we don't need 0 check. Seems like I did a premature optimization here 😢 . Nice catch on the PR though.

**Comment by yaegassy** [comment]
I will leave the details of the change to @prabirshrestha 😄

**Comment by yaegassy** [comment]
This PR is closed. 😄

**Pull request closed without merging**